modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -337,6 +337,10 @@ private void failQueuedWrites() {
         while ((queuedWrite = queuedWrites.poll()) != null) {
             queuedWrite.failAsClosedChannel();
         }
+        if (currentChunkedWrite != null) {
+            currentChunkedWrite.combiner.finish(currentChunkedWrite.onDone());
+            currentChunkedWrite = null;
+        }
```

Comment:
Since we did not set `currentChunkedWrite` back to `null`. Does it mean the assertion of `assert currentChunkedWrite == null;` woudl have been triggered if the channel is reused for another chunked request?

The `close` method below completes `currentChunkedWrite.onDone` directly without going through the combiner. I suppose we could also do the same here. But going through combiner gives us better error message?

Reply:
Once we get here the channel is inactive so we won't be handling any more requests.

Good point about the handling in `close()` tho, maybe we should close it exceptionally. It doesn't really matter I think, the exception doesn't go anywhere. I'm a little suspicious about not using the combiner, I think that means we might in theory close the response while some of the pages it created are still in use, but I can't see an obviously-bad consequence of this either.